Catappult/appcoins-sdk-ios
# Feature/APP-3120_payment_method_selection_second_screen

**What does this PR do?**

   This PR adds a dedicated view for browsing and selecting payment methods, enhancing the payment experience by providing users with a focused interface. Upon selecting a payment method, users are seamlessly redirected back to the main payment screen with their choice pre-selected, aligning with best practices for clarity and ease of use in the payment process. 

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] BottomSheetView.swift
- [ ] PurchaseBottomSheet.swift
- [ ] SelectedPaymentMethodView.swift
- [ ] PaymentMethodChoiceView.swift
- [ ] BottomSheetViewModel.swift

**How should this be manually tested?**

  Run the Trivial Drive feature and tap on the “Select Payment Method” button. Choose a payment method and verify that the flow correctly returns to the main bottom sheet with the selected payment method pre-selected.

**What are the relevant tickets?**

  Tickets related to this pull-request: APP-3120

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 

**Code Review Checklist**
